util/tracer_replayer.h

```diff
@@ -0,0 +1,88 @@
+// Copyright (c) 2011-present, Facebook, Inc.  All rights reserved.
+// This source code is licensed under the BSD-style license found in the
+// LICENSE file in the root directory of this source tree. An additional grant
+// of patent rights can be found in the PATENTS file in the same directory.
+
+#pragma once
+#include <condition_variable>
+#include <memory>
+#include <mutex>
+#include <queue>
+#include <unordered_map>
+#include <string>
+#include "rocksdb/trace_reader_writer.h"
+#include "db/column_family.h"
+#include "port/port.h"
+#include "util/coding.h"
+
+namespace rocksdb {
+
+class DBImpl;
+class Slice;
+
+enum TraceType : char {
+    kTraceStart = 0,
+    kTraceEnd,
+    kTraceWrite,
+    kTraceRead,
+    kTraceIterator,
+    };
+
+using TraceRecord = std::pair<std::string, std::string>;
+
+class Tracer {
```

Comment:
Good idea. For trace, I think there is no more than 1 instance. But for replay, I am not quite sure. There is possibly some weird use cases that ppl may try double workload? We could discuss this.